tests/sentry/workflow_engine/migration_helpers/test_migrate_rule.py

```diff
@@ -406,3 +295,294 @@ def test_delete_issue_alert(self):
+            type=Condition.REAPPEARED_EVENT, comparison=True, condition_result=True
+        ).exists()
+        assert conditions.filter(
+            type=Condition.REGRESSION_EVENT, comparison=True, condition_result=True
+        ).exists()
+
+        if_dcg = WorkflowDataConditionGroup.objects.get(workflow=workflow).condition_group
+        assert if_dcg.logic_type == DataConditionGroup.Type.ANY_SHORT_CIRCUIT
+        filters = DataCondition.objects.filter(condition_group=if_dcg)
+        assert filters.count() == 1
+        assert filters.filter(
+            type=Condition.AGE_COMPARISON,
+            comparison={
+                "comparison_type": AgeComparisonType.OLDER,
+                "value": 10,
+                "time": "hour",
+            },
+            condition_result=True,
+        ).exists()
+
+        dcg_actions = DataConditionGroupAction.objects.get(condition_group=if_dcg)
+        action = dcg_actions.action
+        assert action.type == Action.Type.SLACK
+
+    def test_run_no_double_migrate(self):
+        IssueAlertMigrator(self.issue_alert, self.user.id).run()
+
+        # there should be only 1
+        issue_alert_workflow = AlertRuleWorkflow.objects.get(rule=self.issue_alert)
+        issue_alert_detector = AlertRuleDetector.objects.get(rule=self.issue_alert)
+        Workflow.objects.get(id=issue_alert_workflow.workflow.id)
+        Detector.objects.get(id=issue_alert_detector.detector.id)
+
+    def test_run__detector_exists(self):
+        project_detector = self.create_detector(project=self.project)
+        IssueAlertMigrator(self.issue_alert, self.user.id).run()
+
+        # does not create a new error detector
+
+        detector = Detector.objects.get(project_id=self.project.id)
+        assert detector == project_detector
+
+    def test_run__with_conditions(self):
+        issue_alert = self.create_project_rule(
+            condition_data=self.conditions,
+            action_match="all",
+            filter_match="any",
+            action_data=self.action_data,
+        )
+
+        IssueAlertMigrator(issue_alert, self.user.id).run()
+        assert DataCondition.objects.all().count() == 1
+        dc = DataCondition.objects.get(type=Condition.EVENT_UNIQUE_USER_FREQUENCY_COUNT)
+        assert dc.comparison == {
+            "interval": "1h",
+            "value": 50,
+            "filters": self.expected_filters,
+        }
+
+    def test_run__every_event_condition__any(self):
+        conditions = [
+            {"id": EveryEventCondition.id},
+            {"id": RegressionEventCondition.id},
+        ]
+        issue_alert = self.create_project_rule(
+            condition_data=conditions,
+            action_match="any",
+            filter_match="any",
+            action_data=self.action_data,
+        )
+
+        IssueAlertMigrator(issue_alert, self.user.id).run()
+        assert DataCondition.objects.all().count() == 1
+        dc = DataCondition.objects.get(type=Condition.REGRESSION_EVENT)
```

Comment:
I think the count = 1 is slightly unnecessary as `.get()` would fail if there were more than 1 result

Reply:
i need to make sure there aren't any other `DataCondition` objects created, there could have been something of a different `type`